src/lib/y2partitioner/widgets/mkfs_optiondata.rb

```diff
@@ -0,0 +1,465 @@
+          label:       N_("Number of &FATs"),
+          values:      %w(auto 1 2),
+          default:     "auto",
+          mkfs_option: "-f",
+          # help text, richtext format
+          help:        N_(
+            "Specify the number of file allocation tables. " \
+            "The default is 2."
+          )
+        },
+
+        {
+          fs:          %i(vfat),
+          widget:      :MkfsInputField,
+          label:       N_("Root &Dir Entries"),
+          default:     "auto",
+          validate:    lambda do |x|
+            (x.match?(/^\d+$/) && x.to_i >= 16) || x == "auto"
+          end,
+          mkfs_option: "-r",
+          error:       N_(
+            "The minimum number of entries is 16."
+          ),
+          # help text, richtext format
+          help:        N_(
+            "Select the number of entries available in the root directory. " \
+            "Choose a value that is at least 16, or 'auto'. " \
+            "Note that the value is a lower limit and the actual root directory may be larger."
+          )
+        }
+      ]
+
+      # Remember option data.
+      #
+      # @param option [Hash]
+      #
+      def initialize(option)
+        @value = option
+        textdomain "storage"
+      end
+
+      # Read current option value from file system class.
+      #
+      # @param filesystem [Y2Storage::Filesystems]
+      #
+      # @return [String, Boolean]
+      #
+      def get(filesystem)
+        # current filesystem settings
+        fs_str = filesystem.send program + "_options"
+
+        value = default
+
+        # find option args in current settings; if it's a non-boolean
+        # option get the value from the match
+        m = fs_str.match(/(^|\s)#{Regexp.escape(option_str)}(\S*)/)
+        value = boolean? ? !default : m[2] if m
+
+        value
+      end
+
+      # Set new option value in file system class.
+      #
+      # @param filesystem [Y2Storage::Filesystems]
+      # @param val [String, Boolean]
+      #
+      # @return [nil]
+      #
+      def set(filesystem, val)
+        # current filesystem settings
+        fs_str = filesystem.send program + "_options"
+
+        # remove the currently set option value
+        fs_str.gsub!(/(^|\s+)#{Regexp.escape(option_str)}[^-]*/, "")
+
+        # if it's not the default value, add option string
+        if val != default
+          fs_str << " #{option_str}"
+          fs_str << val if !boolean?
+        end
+
+        # beautify
+        fs_str.strip!
+
+        # store new option args
+        filesystem.send program + "_options=", fs_str
+      end
+
+      # Validate option value.
+      #
+      # This calls the validation function if there is one defined _and_ an
+      # error message exists. Else it just returns true.
+      #
+      # @param val [String, Boolean]
+      #
+      # @return [Boolean]
+      #
+      def validate?(val)
+        return true unless validate && error
```

Comment:
Hmmm, `#error` will always return something true-ish, isnt'it? As it's defined below, it will at least return `"\n\n"`